src/openklant/management/commands/migrate_to_v2_phonenumbers.py

```diff
@@ -0,0 +1,219 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2024 Dimpact
+
+
+import os
+from dataclasses import fields as dataclass_fields
+from typing import Any, Tuple
+from urllib.parse import parse_qs, urlparse
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured, ValidationError
+from django.core.management import CommandError
+from django.core.management.base import BaseCommand, CommandParser
+
+from rest_framework.fields import URLValidator
+from rest_framework.reverse import reverse_lazy
+
+from openklant.components.klantinteracties.constants import SoortDigitaalAdres
+from openklant.components.klantinteracties.models import (
+    DigitaalAdres,
+    Partij,
+)
+from openklant.components.token.models import TokenAuth
+from openklant.migration.client import Client
+from openklant.migration.utils import (
+    MIGRATION_TOKEN_IDENTIFIER,
+    _generate_dummy_token,
+    generate_jwt_token,
+    logger,
+)
+from openklant.migration.v1.client import LegacyOpenKlantClient
+from openklant.migration.v1.data import Klant
+from openklant.migration.v2.client import OpenKlantClient
+
+KLANT_FIELDS = {field.name: field for field in dataclass_fields(Klant)}
+
+DIGITALE_ADDRESSEN_PATH = reverse_lazy(
+    "klantinteracties:digitaaladres-list",
+    kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+)
+
+PARTIJEN_PATH = reverse_lazy(
+    "klantinteracties:partij-list",
+    kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+)
+
+
+def _retrieve_klanten(url: str, access_token: str) -> Tuple[list[Klant], str | None]:
+    klanten_path = "/klanten/api/v1/klanten"
+
+    _url = urlparse(url)
+    _params = parse_qs(_url.query)
+
+    client = LegacyOpenKlantClient(f"{_url.scheme}://{_url.netloc}", access_token)
+    response_data = client.retrieve(
+        _url.path or klanten_path, params={**_params, "expand": "digitaleAdressen"}
```

Comment:
Sorry, I was wrong about this, I forgot that this is the API call to v1 and `digitaleAddressen` don't exist there, so this can be removed :facepalm: 

```suggestion
        _url.path or klanten_path, params=_params
```